zetteldeft.el

```diff
@@ -326,52 +331,68 @@ Don't forget to add `\\n' at the beginning to start a new line."
-     "" baseName)))
+  (let ((deft-use-filename-as-title nil))
+    (deft-parse-title zdFile (with-temp-buffer
+      (insert-file-contents zdFile nil nil nil t)
+       (buffer-string)))))
 
 (defun zetteldeft-file-rename ()
-  "Rename the current file via the deft function.
-Use this on files in the `deft-directory'."
+  "Change current file's title, and use the new title to rename the file."
   (interactive)
   (zetteldeft--check)
-    (let ((old-filename (buffer-file-name))
-          (deft-dir (file-name-as-directory deft-directory))
-          new-filename old-name new-name)
-      (when old-filename
-        (setq old-name (deft-base-filename old-filename))
-        (setq new-name (read-string
-                        (concat "Rename " old-name " to (without extension): ")
-                        old-name))
-        (setq new-filename
-              (concat deft-dir new-name "." deft-default-extension))
-        (rename-file old-filename new-filename)
-        (deft-update-visiting-buffers old-filename new-filename)
-        (zetteldeft-update-title-in-file)
-        (deft-refresh))))
-
-(defun zetteldeft-update-title-in-file ()
+  (let ((old-filename (buffer-file-name)))
+    (when old-filename
+      (let* ((old-title (zetteldeft--lift-file-title old-filename))
+	     (prompt-text (concat "Change " old-title " to: "))
+	     (new-name (read-string prompt-text old-title))
+	     (old-id (zetteldeft--lift-id old-filename))
+	     (new-filename (deft-absolute-filename (concat
+						    old-id
+						    zetteldeft-id-filename-separator
+						    new-name))))
+	(rename-file old-filename new-filename)
+	(deft-update-visiting-buffers old-filename new-filename)
+        (zetteldeft-update-title-in-file new-name)
+	(deft-refresh)))))
+
+(defun zetteldeft-update-title-in-file (title)
   "Update the title of the current file, if present.
-Does so by looking for `zetteldeft-title-prefix'."
+Does so by looking for `zetteldeft-title-prefix'.
+Replaces current title with TITLE."
   (save-excursion
     (let ((zetteldeft-title-suffix ""))
       (goto-char (point-min))
       (when (re-search-forward (regexp-quote zetteldeft-title-prefix) nil t)
         (delete-region (line-beginning-position) (line-end-position))
-        (zetteldeft--insert-title)))))
+        (zetteldeft--insert-title title)))))
+
+(defun zetteldeft-file-rename-full ()
+  "Rename the current file.
+Allows modification of entire filename."
+  (interactive)
+  (zetteldeft--check)
+  (let ((old-filename (buffer-file-name)))
+    (when old-filename
+      (let* ((old-name (file-name-nondirectory old-filename))
+	     (prompt-text (concat "Rename " old-name " to: "))
+	     (new-name (read-string prompt-text old-name))
+	     (deft-dir (file-name-as-directory deft-directory))
+	     (new-filename (concat deft-dir new-name)))
+	(rename-file old-filename new-filename)
+	(deft-update-visiting-buffers old-filename new-filename)
+	(deft-refresh)))))
```

Comment:
I used this function recently to modify some filenames where the ID wasn't unique - I'd created a few notes in the same minute. You're right, `deft-rename-file` would have worked. I like having a function to rename the file in the current buffer instead of from the deft buffer, but this could live in my emacs config. Happy to remove if you think it should go.

Reply:
I do think this belongs in your config rather than in zetteldeft, yes.